Libraries/src/Amazon.Lambda.Annotations.SourceGenerator/Writers/CloudFormationWriter.cs

```diff
@@ -0,0 +1,323 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using Amazon.Lambda.Annotations.APIGateway;
+using Amazon.Lambda.Annotations.SourceGenerator.Diagnostics;
+using Amazon.Lambda.Annotations.SourceGenerator.FileIO;
+using Amazon.Lambda.Annotations.SourceGenerator.Models;
+using Amazon.Lambda.Annotations.SourceGenerator.Models.Attributes;
+using Microsoft.CodeAnalysis;
+using Newtonsoft.Json.Linq;
+using YamlDotNet.RepresentationModel;
+
+namespace Amazon.Lambda.Annotations.SourceGenerator.Writers
+{
+    /// <summary>
+    /// This class contains methods to manipulate the AWS serverless template.
+    /// It takes the metadata captured by <see cref="AnnotationReport"/> and writes it to the AWS SAM template.
+    /// <see href="https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-resource-function.html">see here</see> to know more about configurable properties for AWS::Serverless::Function
+    /// <see href="https://github.com/aws/aws-lambda-dotnet/blob/master/Libraries/test/TestServerlessApp/serverless.template">see here</see> for an actual serverless template.
+    /// </summary>
+    public class CloudFormationWriter : IAnnotationReportWriter
+    {
+        private readonly IFileManager _fileManager;
+        private readonly IDirectoryManager _directoryManager;
+        private readonly ITemplateWriter _templateWriter;
+        private readonly CloudFormationTemplateFormat _templateFormat;
+        private readonly IDiagnosticReporter _diagnosticReporter;
+
+        public CloudFormationWriter(IFileManager fileManager, IDirectoryManager directoryManager, CloudFormationTemplateFormat templateFormat, IDiagnosticReporter diagnosticReporter)
+        {
+            _fileManager = fileManager;
+            _directoryManager = directoryManager;
+            _diagnosticReporter = diagnosticReporter;
+            _templateFormat = templateFormat;
+
+            if (templateFormat == CloudFormationTemplateFormat.Json)
+                _templateWriter = new JsonWriter();
+            else
+                _templateWriter = new YamlWriter();
+        }
```

Comment:
Might not matter much in this specific case, but generally would prefer to see this take in an `ITemplateWriter` via the constructor, rather than the enum. That leaves us more options when it comes to mocking/testing.

Reply:
Changed this to accept `ITemplateWriter` via the constructor.